packages/cli/config/jestCachingModuleLoader.js

```diff
@@ -21,13 +21,18 @@ const fileTransformCache = new Map();
 const scriptTransformCache = new Map();
 
 let runtimeGeneration = 0;
+let isWatchMode;
 
 module.exports = class CachingJestRuntime extends JestRuntime {
   // Each Jest run creates a new runtime, including when rerunning tests in
   // watch mode. This keeps track of whether we've switched runtime instance.
   __runtimeGeneration = runtimeGeneration++;
 
   transformFile(filename, options) {
+    if (!isWatchMode) {
+      return super.transformFile(filename, options);
+    }
```

Comment:
The purpose of this runtime is also to override the default behavior in Jest where transforms are cached per project rather than globally. I'm thinking we want to keep that, but could skip the time check during watch mode if it's worth it.

Reply:
Ah, perhaps I'm misunderstanding, but isn't that already the case?
I think that [`_fileTransforms`](https://github.com/facebook/jest/blob/main/packages/jest-runtime/src/index.ts#L1511-L1516) are scoped to the current process ("global"), not per-project.

The [`_cache` of `ScriptTransformer`](https://github.com/facebook/jest/blob/main/packages/jest-transform/src/ScriptTransformer.ts#L706) is scoped per-project (due to the implementation of `getScriptCacheKey()`) IIRC.

So, `fileTransformCache` is redundant if we're not doing `mtime` checks, right?

Reply:
Ah nice, this same caching was added to Jest a while ago ([old code](https://github.com/facebook/jest/blob/343532a21f640ac2709c4076eef57e52279542e1/packages/jest-runtime/src/index.ts#L1064-L1081)). Makes more sense in the Jest end now.

Let's :shipit:! 👍